src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1744,6 +1745,150 @@ class Simplex_tree {
     }
   }
 
+ public:
+   /** @private @brief Returns the serialization required buffer size.
+   * 
+   * @return The exact serialization required size in number of bytes.
+   * 
+   * @warning It is meant to return the same size with the same SimplexTreeOptions and on a computer with the same
+   *   architecture.
+   */
+  std::size_t get_serialization_size() {
+    const std::size_t vh_byte_size = sizeof(Vertex_handle);
+    const std::size_t fv_byte_size = SimplexTreeOptions::store_filtration ? sizeof(Filtration_value) : 0;
+    const std::size_t buffer_byte_size = vh_byte_size + num_simplices() * (fv_byte_size + 2 * vh_byte_size);
+#ifdef DEBUG_TRACES
+      std::clog << "Gudhi::simplex_tree::get_serialization_size - buffer size = " << buffer_byte_size << std::endl;
+#endif  // DEBUG_TRACES
+    return buffer_byte_size;
+  }
+  
+  /** @private @brief Serialize the Simplex tree - Flatten it in a user given array of char
+   * 
+   * @param[in] buffer An array of char allocated with enough space (cf. Gudhi::simplex_tree::get_serialization_size)
+   * @param[in] buffer_size The buffer size.
+   * 
+   * @exception std::invalid_argument If serialization does not match exactly the buffer_size value.
+   * 
+   * @warning Serialize/Deserialize is not portable. It is meant to be read in a Simplex_tree with the same
+   * SimplexTreeOptions and on a computer with the same architecture.
+   */
+  /* Let's take the following simplicial complex as example:         */
+  /* (vertices are represented as letters to ease the understanding) */
+  /*  o---o---o */
+  /*  a   b\X/c */
+  /*        o   */
+  /*        d   */
+  /* The simplex tree is: */
+  /* a o  b o     c o   d o   */
+  /*   |    |\      |         */
+  /* b o  c o o d   o d       */
+  /*        |                 */
+  /*      d o                 */
+  /* The serialization is (without filtration values that comes right after vertex handle value):                    */
+  /* 04(number of vertices)0a 0b 0c 0d(list of vertices)01(number of [a] children)0b([a,b] simplex)                  */
+  /* 00(number of [a,b] children)02(number of [b] children)0c 0d(list of [b] children)01(number of [b,c] children)   */
+  /* 0d(list of [b,c] children)00(number of [b,c,d] children)00(number of [b,d] children)01(number of [c] children)  */
+  /* 0d(list of [c] children)00(number of [b,d] children)00(number of [d] children)                                  */
+  /* Without explanation and with filtration values:                                                                 */
+  /* 04 0a F(a) 0b F(b) 0c F(c) 0d F(d) 01 0b F(a,b) 00 02 0c F(b,c) 0d F(b,d) 01 0d F(b,c,d) 00 00 01 0d F(c,d) 00 00 */
+  void serialize(char* buffer, const std::size_t buffer_size) {
```

Comment:
`buffer_size` is there only to verify that it matches? Hmm... ok.
An alternative would have been for serialize to return `buffer_end` (like serialize_trivial), and leave it to the caller to do the verifications they want, but that wouldn't matter for our current use, so this version is fine.